Source/CarthageKit/VersionFile.swift

```diff
@@ -1,45 +1,36 @@
 import Foundation
 import Runes
-import Argo
-import Curry
 import ReactiveSwift
 import ReactiveTask
 import Result
 import XCDBLD
 
-struct CachedFramework {
+struct CachedFramework: Codable {
+	enum CodingKeys: CodingKey, String {
+		case name = "name"
+		case hash = "hash"
+	}
+
 	let name: String
 	let hash: String
-
-	static let nameKey = "name"
-	static let hashKey = "hash"
-
-	func toJSONObject() -> Any {
-		return [
-			CachedFramework.nameKey: name,
-			CachedFramework.hashKey: hash,
-		]
-	}
 }
 
-extension CachedFramework: Argo.Decodable {
-	static func decode(_ json: JSON) -> Decoded<CachedFramework> {
-		return curry(self.init)
-			<^> json <| CachedFramework.nameKey
-			<*> json <| CachedFramework.hashKey
+struct VersionFile: Codable {
+	enum CodingKeys: CodingKey, String {
+		case commitish = "commitish"
```

Comment:
is there a reason why you duplicate the case into a string when it is the same? Like:
```swift 
case commitish = "commitish"
```
instead of 
```swift 
case commitish 
```

Reply:
I think the explicitness is a virtue and durable against some refactorings (renaming).

Reply:
Ok. Makes sense